rust/src/internal/core.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::internal::adv::AdvertisementDataValue;
+use lazy_static::lazy_static;
+use nom::combinator;
+use std::fmt;
+
+lazy_static! {
+    static ref BASE_UUID: [u8; 16] = hex::decode("0000000000001000800000805F9B34FB")
+        .unwrap()
+        .try_into()
+        .unwrap();
+}
+
+/// 16-bit UUID
+#[derive(PartialEq, Eq, Hash, Clone, Copy)]
+pub struct Uuid16 {
+    /// Big-endian bytes
+    be_bytes: [u8; 2],
+}
+
+impl Uuid16 {
+    /// Construct a UUID from little-endian bytes
+    pub fn from_le_bytes(mut le_bytes: [u8; 2]) -> Self {
+        le_bytes.reverse();
+        Self::from_be_bytes(le_bytes)
+    }
+
+    /// Construct a UUID from big-endian bytes
+    pub const fn from_be_bytes(be_bytes: [u8; 2]) -> Self {
+        Self { be_bytes }
+    }
+
+    /// The UUID in big-endian bytes form
+    pub fn as_be_bytes(&self) -> [u8; 2] {
+        self.be_bytes
+    }
+
+    /// The UUID in little-endian bytes form
+    pub fn as_le_bytes(&self) -> [u8; 2] {
+        let mut uuid = self.be_bytes;
+        uuid.reverse();
+        uuid
+    }
+
+    pub(crate) fn parse_le(input: &[u8]) -> nom::IResult<&[u8], Self> {
+        combinator::map_res(nom::bytes::complete::take(2_usize), |b: &[u8]| {
```

Comment:
These are intentional to show a hint of where the functions come from -- though the `nom` could certainly be lopped off, I think.

Reply:
I personally would find this easier to read without the long namespaces. But I've also done far less parsing than you have, so knowing when its important to include these namespaces goes a little over my head (aside from when needing to disambiguate, like if you were using `streaming::take` and `complete::take` in this same module).

Reply:
Yep, that's pretty much it: if you just `take`, are you taking bytes? bits? 

In general I've found it helpful for reader context to leave a bit of the module path present for things outside the current crate, or in unusual circumstances, the current module.

Reply:
Oh, now I remember... I can't chop off `nom` because we already have the `bytes` crate in scope, and introducing different names at use-time seems even more confusing.